pandas/core/generic.py

```diff
@@ -5907,6 +5908,137 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_type(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        result = self
+        target_int_dtype = "Int64"
+        if is_platform_32bit():
+            target_int_dtype = "Int32"
+
+        if is_object_dtype(dtype):
+            new_dtype = lib.infer_dtype(self)
+            if (
+                new_dtype != "string"
+                and new_dtype != "boolean"
+                and new_dtype != "integer"
+            ):
```

Comment:
I think you could shorten this as `new_dtype not in {"string", "boolean", "integer"}`

Reply:
> 
> I would maybe only use nullable_**d**types (with the "d"), since that's the term that is used elsewhere in APIs in pandas (eg `dtypes` property, `dtype=` keyword, etc).

I like that idea.  Will be in my next commit.
> 
> Something else I was wondering: does this need to be a method?
> Yes, a method is certainly more discoverable. But to me this doesn't feel like a typical operation, it's also kind of a temporary thing to try out (in awaiting it to be the default at some point), so a top-level function `pd.as_nullable_dtypes(..)` might also be fine?

I'm indifferent here, but I have a sense that this method may be around a while as we convert other things to use `pd.NA`.  So I'm -1 on putting it at top level, but will defer to others if we want to change it.